js/foam/core/Enum.js

```diff
@@ -35,6 +35,29 @@ CLASS({
       }
     },
     {
+      /* Not ready to be used yet. For it to work, we need to change the way the
+       * build happens and bundle the generated files as part of the android
+       * project instead of as their own library to avoid a circular dependency.
+       */
+      name: 'androidResource',
+    },
+    {
+      name: 'messages',
+      defaultValueFn: function() {
```

Comment:
A defaultValueFn runs the function every time the property is accessed.  This can be slow and create a lot of garbage if you construct stuff in that function like you are doing here.

I would suggest you use lazyFactory: instead.  lazyFactory runs only first time the property is accessed and then stores the returned value for subsequent accesses.